src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -184,6 +193,28 @@ internal PackageSourceMoniker SelectedSource
             }
 
             _missingPackageStatus = false;
+
+            var gen = PackageList._list.ItemContainerGenerator;
+            gen.StatusChanged += Gen_StatusChanged;
+        }
+
+        private void PackageList_LoadItemsCanceled(object sender, EventArgs e)
+        {
+            if (!_loadCts.IsCancellationRequested)
```

Comment:
This cancel handler is hit multiple times since multiple worker threads see the cancelled token.
Therefore, only "stop and retry" the PMControl's loading token once.